resources/assets/lib/user-card.tsx

```diff
@@ -0,0 +1,207 @@
+        <div className='usercard__icon'>
+          <a href={laroute.route('rankings', { mode: 'osu', type: 'performance', country: this.user.country_code })}>
+            <FlagCountry country={ this.user.country }/>
+          </a>
+        </div>
+
+        {
+          this.user.is_supporter ?
+          <div className='usercard__icon'>
+            <a className='usercard__link-wrapper' href={laroute.route('support-the-game')}>
+              <SupporterIcon smaller={true} />
+            </a>
+          </div> : null
+        }
+
+        <div className='usercard__icon'>
+          <FriendButton userId={this.user.id} />
+        </div>
+
+        {
+          this.canMessage ?
+          <div className='usercard__icon'>
+            <a className='user-action-button user-action-button--message'
+              href={laroute.route('messages.users.show', { user: this.user.id })}
+              title={osu.trans('users.card.send_message')}
+            >
+              <i className='fas fa-envelope'></i>
+            </a>
+          </div> : null
+        }
+      </div> : null;
+
+    let usercardCss = 'usercard';
+    for (const modifier of this.props.modifiers) {
+      usercardCss += ` usercard--${modifier}`;
+    }
+
+    return (
+      <div className={usercardCss}>
+        { this.renderBackground() }
+
+        <div className='usercard__card'>
+          <div className='usercard__card-content'>
+            { this.renderAvatar() }
+
+            <div className='usercard__metadata'>
+              <div className='usercard__username'>{ this.user.username }</div>
+              { details }
+            </div>
+          </div>
+          { this.renderOnlineStatus() }
+        </div>
+      </div>
+    );
+  }
+
+  renderAvatar() {
+    let avatarSpaceCssClass = 'usercard__avatar-space';
+    if (!this.state.avatarLoaded) {
+      avatarSpaceCssClass += ' usercard__avatar-space--loading';
+    }
+
+    return (
+      <div className={avatarSpaceCssClass}>
+        <div className='usercard__avatar usercard__avatar--loader'>
+          { !this.isUserNotFound ? <div className='la-ball-clip-rotate'></div> : null }
+        </div>
+        {
+          this.isUserLoaded ? <img className='usercard__avatar usercard__avatar--main'
+                                   onError={this.onAvatarLoad} // remove spinner if error
+                                   onLoad={this.onAvatarLoad}
+                                   src={this.user.avatar_url}
+                              />
+                            : null
+        }
+      </div>
+    );
+  }
+
+  renderBackground() {
+    let background: React.ReactNode;
+    let backgroundLink: React.ReactNode;
+
+    if (this.user.cover && this.user.cover.url) {
+      let backgroundCssClass = 'usercard__background';
+      if (!this.state.backgroundLoaded) {
+        backgroundCssClass += ' usercard__background--loading';
+      }
+
+      background =
+        <React.Fragment>
```

Comment:
I actually forgot that `<>` resolves to `<React.Fragment>` but it's a thing I've been using (the short syntax).